lib/src/commands/create/commands/create_subcommand.dart

```diff
@@ -121,13 +121,24 @@ abstract class CreateSubCommand extends Command<int> {
     return Directory(directory);
   }
 
-  /// Gets the project name.
-  String get projectName {
+  /// The project name that the user provided.
+  ///
+  /// Since the project name could be user provided as either:
+  /// 1. A valid package name
+  /// 2. '.', the current working directory (the project assumes the cwd's name)
+  /// this needs to exist to provide a fallback value for [outputDirectory] and
+  /// [projectName].
+  String get _projectName {
     final args = argResults.rest;
     _validateProjectName(args);
     return args.first;
   }
 
+  /// Gets the project name.
+  String get projectName => _projectName == '.'
+      ? path.basename(Directory.current.path)
+      : _projectName;
+
```

Comment:
I think we should rework this. Introducing `.` as a `projectName` and using it to find the name of the directory is kinda confusing because what if I pass `path/to/my/folder` instead?

`.` implies current directory which implicitly also means we can pass any path, which isnt the case. However this is the case for `flutter create` so maybe we should just make it that we can pass any given path and use the last part of that path as the `projectName` and create the template at the given path.

What you think?

Reply:
Hi @wolfenrain, thanks for the positive feedback.

I agree, let's do it the same way as the Flutter tool.

This means we'd remove `--output-directory` as an argument. 

As far as `very_good create <arg>`, here are some requirements that I could implement:
- [ ] Allow relative paths as `arg`.
- [ ] Allow absolute paths as `arg`.
- [ ] Use the last path segment of `arg` as the project name.
- [ ] Remove the `--output-directory` argument.

Let me know if you'd like me to proceed. Thanks for catching my oversight 👍

Reply:
I do think that is a good approach, so `arg` would be any type of path indeed, it would be nice if we can keep backwards compatibility with the `--output-directory`. Maybe by just checking if it is supplied, if it is show that it is deprecated and just take that over whatever was in `arg`, assuming `arg` was a path?

PS: it would be `very_good create <template> <output directory>`

Reply:
Hey @a-wallen just following up here to see if you are planning on tackling this still?